### PR TITLE
Add view and tests for stripe user subscription cancelation.

### PR DESCRIPTION
**Pull request by michal-macioszczyk**
Implementing a new feature to allow customers to cancel their recurring subscription on their profile page. This simply calls the Stripe API to cancel the current payment plan.

**in this PR**
https://trello.com/c/XuSGCflk/33-as-a-user-i-want-to-be-able-to-cancel-my-subscription-from-my-profile-page

**images**
screenshot 2018-11-27 at 10 17 15
screenshot 2018-11-27 at 18 09 02
screenshot 2018-11-27 at 18 09 14

**Notes for QA**
- [ ] Ensure subscription link appears if the recurring_payments feature is enabled.
- [ ] If the user has a subscription ensure the information (card details/amount) is correct.
- [ ] Ensure styling copy and links on the payment management page work and display correctly
- [ ] Ensure cancelling a subscription works and returns a success message

### CODE REVIEW

File: kuma/payments/jinja2/payments/management.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <meta name="description" content="{{ seo_description }}">
+    <meta name="twitter:description" content="{{ seo_description }}">
+    <meta name="twitter:title" content="{{ social_title }}">
+    <meta property="og:title" content="{{ social_title }}">
+{% endblock %}
+
+
+{% block content %}
+    <div>
+        <h2>Subscriptions</h2>
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: This should be `h1` if it's the first heading on the page.

File: kuma/payments/jinja2/payments/management.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <meta property="og:description" content="{{ seo_description }}">
+    <meta name="description" content="{{ seo_description }}">
+    <meta name="twitter:description" content="{{ seo_description }}">
+    <meta name="twitter:title" content="{{ social_title }}">
+    <meta property="og:title" content="{{ social_title }}">
+{% endblock %}
+
+
+{% block content %}
+    <div>
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: What's this `div` doing?

File: kuma/payments/jinja2/payments/management.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        <h2>Subscriptions</h2>
+        {% if request.user.stripe_customer_id and active_subscriptions %}
+            <p>
+                Monthly payment to MDN
+                ${{ stripe_plan_amount }} from your account ending in {{ stripe_card_last4 }}
+                <a href="{{ url('payment_terms') }}">Learn more about subscriptions</a>
+            </p>
+            <form id="id_stripe_management_form" method="post" action="" >
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: This form doesn't have an action. Is it controlled by JS?

File: kuma/payments/jinja2/payments/management.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                {% csrf_token %}
+                <button id="id_stripe_cancel_subscription" name="stripe_cancel_subscription" class="button" type="submit">Cancel your subscription</button>
+            </form>
+        {% else %}
+            <p>
+                Sorry you don't have any active subscription
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: This should be wrapped in a translation tag.

File: kuma/payments/jinja2/payments/management.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <meta property="og:title" content="{{ social_title }}">
+{% endblock %}
+
+
+{% block content %}
+    <div>
+        <h2>Subscriptions</h2>
+        {% if request.user.stripe_customer_id and active_subscriptions %}
+            <p>
+                Monthly payment to MDN
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: All text should be wrapped in a translation tag.

File: kuma/users/jinja2/users/user_edit.html [hunk 6]
```diff
@@ -61,6 +61,11 @@ <h1 class="user-title">{{ edit_user.username }} <a class="user-title-profile" hr
             <label>{{ _('Primary Email') }} <a class="field-management" href="{{ url('account_email') }}">{{ _('Edit email') }}</a></label>
             {{ edit_user.email }}
           </li>
+            {% if request.user.stripe_customer_id %}
+                <li id="field_subscription" class="">
```
Comment (matthewhall) [inline kuma/users/jinja2/users/user_edit.html]: Empty `class`. Do we need it?

File: kuma/users/jinja2/users/user_edit.html [hunk 7]
```diff
@@ -61,6 +61,11 @@ <h1 class="user-title">{{ edit_user.username }} <a class="user-title-profile" hr
...
+                    <label>{{ _('Subscriptions') }} <a class="field-management" href="{{ url('recurring_payment_management') }}">{{ _('Edit payment subscription') }}</a></label>
```
Comment (matthewhall) [inline kuma/users/jinja2/users/user_edit.html]: `label`s should be associated with a form element using the `for` attribute.

File: kuma/payments/jinja2/payments/management.html [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+                ${{ stripe_plan_amount }} from your account ending in {{ stripe_card_last4 }}
+                <a href="{{ url('payment_terms') }}">Learn more about subscriptions</a>
+            </p>
+            <form id="id_stripe_management_form" method="post" action="" >
+                {% csrf_token %}
+                <button id="id_stripe_cancel_subscription" name="stripe_cancel_subscription" class="button" type="submit">Cancel your subscription</button>
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: Button text should be wrapped in a translation tag.

File: kuma/payments/jinja2/payments/management.html (hunk 1 above)
Comment (matthewhall) [reply kuma/payments/jinja2/payments/management.html]: Also the text should be tagged for translation.

File: kuma/payments/jinja2/payments/management.html (hunk 4 above)
Comment (matthewhall) [reply kuma/payments/jinja2/payments/management.html]: Should be written in proper sentence case and with correct punctuation as well. In this case, it should be like this:

```
Sorry, you don't have any active subscription.
```

File: kuma/payments/jinja2/payments/management.html (hunk 1 above)
Comment (joshJarr) [reply kuma/payments/jinja2/payments/management.html]: There's now a header above `<h1>{{_('Your MDN profile')}}</h1>`

File: kuma/payments/jinja2/payments/management.html (hunk 3 above)
Comment (joshJarr) [reply kuma/payments/jinja2/payments/management.html]: The action posts back to the same page, looking at http://w3c.github.io/html/#attr-fs-action
it seems like the `action=""` is invalid and that we should just omit the action attr, I'll do this.

File: kuma/static/styles/components/users/manage-subscriptions.scss [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+    @media #{$mq-small-mobile-and-down} {
+        .column-4 {
+            margin: 1em 0;
+            padding: 0;
+            width: 100%;
+        }
+    }
+
+    header {
```
Comment (matthewhall) [inline kuma/static/styles/components/users/manage-subscriptions.scss]: Select these with a class selector not an element selector. `header` is too broad a selector and could affect other areas of the page or site.

And below.

File: kuma/static/styles/components/users/manage-subscriptions.scss [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+            padding: 0;
+        }
+
+        th,
+        td {
+            padding-right: 3em;
+        }
+
+        th {
+            font-family: zillaslab;
```
Comment (matthewhall) [inline kuma/static/styles/components/users/manage-subscriptions.scss]: Always include fallback fonts if you're loading a custom font-family.

File: kuma/static/styles/submission.scss [hunk 11]
```diff
@@ -139,6 +139,13 @@ p.field-note {
     margin: 0 0 5px;
 }
 
+.subscription-link {
+    @include bidi(((float, right, left),));
```
Comment (matthewhall) [inline kuma/static/styles/submission.scss]: Is the syntax of this correct? It looks malformed.

File: kuma/users/jinja2/users/user_edit.html [hunk 12]
```diff
@@ -51,6 +51,7 @@ <h1 class="user-title">{{ edit_user.username }} <a class="user-title-profile" hr
       <fieldset class="section notitle" id="personal">
         <ul>
           <li id="field-beta" class="field">
+            <h2> {{_('User Information')}} </h2>
```
Comment (matthewhall) [inline kuma/users/jinja2/users/user_edit.html]: Please remove the spaces around the template tag.

File: kuma/users/jinja2/users/user_edit.html [hunk 13]
```diff
@@ -85,9 +86,11 @@ <h1 class="user-title">{{ edit_user.username }} <a class="user-title-profile" hr
           </li>
         </ul>
       </fieldset>
-
+        <div class="section notitle" id="subscriptions">
+          <h2>{{ _('Subscriptions') }} <a class="subscription-link" href="{{ url('recurring_payment_management') }}">{{ _('Edit payment subscription') }}</a></h2>
+        </div>
       <fieldset class="section" id="users">
-        <legend><b>{{ _('My Profiles') }}</b></legend>
+        <legend><h2>{{ _('My Profiles') }}</h2></legend>
```
Comment (matthewhall) [inline kuma/users/jinja2/users/user_edit.html]: Is `h2` the correct order here? Seems we have a heading above this which is also `h2`. Shouldn't `h3` be the correct one?

File: kuma/payments/jinja2/payments/management.html [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+                <a href="{{ url('users.user_edit', username=user.username) }}" class="button neutral back">{% include 'includes/icons/arrows/arrow-left.svg' %} {{_('Back to profile')}}</a>
+            </div>
+            <aside class="column-4">
+                <h4>{{_('Help')}}</h4>
+                <ul>
+                    <li>
+                        <a href="{{ url('recurring_payment_initial') }}" class="cta-link full-width">{{_('Monthly subscriptions FAQ')}} {% include 'includes/icons/arrows/arrow.svg' %}</a>
+                    </li>
+                    <li>
+                        <a href="{{ support_mail_link }}" class="cta-link cta-link__email full-width">{{ support_mail }}{% include 'includes/icons/general/envelope.svg' %}</a>
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: How come we're now adopting BEM? `cta-link__email`. I don't recall seeing it used elsewhere, but might be wrong.

File: kuma/payments/jinja2/payments/management.html [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+                        <td>02/19</td>
+                    </tr>
+                </table>
+                {% if not cancel_request %}
+                    <form id="id_stripe_management_form" method="post">
+                        {% csrf_token %}
+                        <button class="button negative" type="button" tooggle-confirmation>{{_('Cancel your subscription')}}</button>
+                        <div id="delete-confirmation" class="hidden">
+                            <div class="confirmation column-8 alert--danger">
+                                <div>
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: Do we need this `div`?

File: kuma/payments/jinja2/payments/management.html [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+                                </div>
+                            </div>
+                        </div>
+                    </form>
+                {% endif %}
+            </section>
+        {% else %}
+            <section>
+                <h3>{{_('Monthly Subscription')}}</h3>
+                <div class="table">
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/management.html]: Should this not be a `table` element?

File: kuma/static/js/payments-manage.js [hunk 17]
```diff
@@ -0,0 +1,25 @@
+
+(function(win, doc) {
+    'use strict';
+
+    var path = win.location.pathname;
+    if (!path.includes('/payments/recurring/management')) {
+        return;
+    }
+
+    var toggleConfirmationButtons = Array.from(doc.querySelectorAll('button[tooggle-confirmation]'));
```
Comment (matthewhall) [inline kuma/static/js/payments-manage.js]: Typo: `tooggle-confirmation`

File: kuma/static/styles/submission.scss [hunk 18]
```diff
@@ -139,6 +139,13 @@ p.field-note {
...
+    font-size: 1rem;
+    font-weight: normal;
+    line-height: 3em;
```
Comment (matthewhall) [inline kuma/static/styles/submission.scss]: Please order these alphabetically.

File: kuma/static/styles/submission.scss (hunk 11 above)
Comment (joshJarr) [reply kuma/static/styles/submission.scss]: I'm reusing their mixin here, it seems to work; but I agree it is confusing. 
I'm using it as they do across site.

**Pull request closed without merging**